test/parallel_api/ranges/std_ranges_find_first_of.pass.cpp

```diff
@@ -0,0 +1,34 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "std_ranges_test.h"
+
+std::int32_t
+main()
+{
+#if _ENABLE_STD_RANGES_TESTING
+    using namespace test_std_ranges;
+    namespace dpl_ranges = oneapi::dpl::ranges;
+
+    auto find_first_of_checker = TEST_PREPARE_CALLABLE(std::ranges::find_first_of);
+
+    test_range_algo<0, int, data_in_in>{medium_size}(dpl_ranges::find_first_of, find_first_of_checker, binary_pred);
```

Comment:
Could you also add a case when nothing is found?

Reply:
see find_first_of.pass.cpp
The same implementation pattern is used for iterator and range based implementation.

Reply:
Looking at the implementation, I see that there is not much additional code, but I still recommend covering that - it is a very basic case, which can be an additional verification that the underlying patter is called the right way.

I prefer covering the such very basic cases. It does not block the PR, but I strongly recommend testing it.